drafts/str4d-overwinter-activation/draft1.rst

```diff
@@ -0,0 +1,196 @@
+::
+
+  ZIP: ???
+  Title: Network Upgrade Mechanism
+  Author: Jack Grigg <jack@z.cash>
+  Comments-Summary: No comments yet.
+  Category: Process
+  Created: 2018-01-08
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST", "MUST NOT", "SHOULD", and "MAY" in this document are to be interpreted as described in
+RFC 2119.
+
+
+Abstract
+========
+
+This proposal defines a mechanism for coordinating upgrades of the Zcash network, in order to remove ambiguity
+about when network upgrades will activate, provide defined periods in which users should upgrade their local
+software, and minimize the risks to both the upgrading network and those users opting out of the changes.
+
+
+Motivation
+==========
+
+Zcash is a *consensual currency*: nobody is ever going to force someone to use a specific version or a
+specific branch of Zcash. [#consensual-currency]_ As such, different sub-communities will always have the
+freedom to choose different variants or branches which offer different design trade-offs.
+
+The current Zcash software includes an *auto-senescence* feature, causing nodes running a particular version
+to automatically shut down 16 weeks after that version was released. This was implemented for several reasons
+[#release-lifecycle]_:
+
+- It gives the same systemic advantage of removing old software as auto-upgrade behavior.
+
+- It requires users to individually choose one of the following options:
+
+  - Upgrade to a more recent software release from the main network.
+
+  - Upgrade to an alternative release.
+
+  - Ignore the "deprecation" and keep running the older software.
+
+Developers can rely on this cadence for coordinating network upgrades. Once the last pre-upgrade software
+version has been deprecated, they can reasonably assume that all node operators on the network either support
+the upgraded rules, or have explicitly chosen not to follow them.
+
+However, this behaviour is not sufficient for performing network upgrades. A globally-understood on-chain
+activation mechanism is necessary so that nodes can unambiguously know at what point the changes from an
+upgrade come into effect (and can enforce consensus rule changes, for example).
+
+
+Specification
+=============
+
+The following constants are defined for every network upgrade:
+
+BRANCH_ID
```

Comment:
See earlier comment about definitions re: why BRANCH_ID and not FORK_ID.

Reply:
Agree it is important to define terms precisely and use precisely the right term for the name of this constant.

Reply:
See the definitions I added for "branch" and "fork" - they are not quite interchangeable, and I believe that `BRANCH_ID` is more precise here.